Influxd startup fails eventually with fatal error: runtime: address space conflict

__System info:__ InfluxDB 1.5.2 & 1.6.1 on FreeBSD 10.4

__Steps to reproduce:__

1. Create a large database (this is about 1 Tb)
2. Stop influxd
3. Start influxd again

Influxd starts going through all tsm files and eventually spits out:

Aug 30 15:08:27 xserver influxd: ts=2018-08-30T12:08:27.217423Z lvl=info msg="Opened shard" log_id=0AEgKNG0000 service=store trace_id=0AEgKOP0000 op_name=tsdb_open index_version=tsi1 path=/va
r/db/influxdb/data/xxxxx/autogen/218 duration=73352.407ms
Aug 30 15:08:27 xserver influxd: runtime: address space conflict: map(0xc460a70000) = 0xcc6f190000(err 0)
Aug 30 15:08:27 xserver influxd: fatal error: runtime: address space conflict
Aug 30 15:08:27 xserver influxd:
Aug 30 15:08:27 xserver influxd: runtime stack:
Aug 30 15:08:27 xserver influxd: runtime.throw(0xe01d8e, 0x1f)
Aug 30 15:08:27 xserver influxd:        /usr/local/go/src/runtime/panic.go:616 +0x81
Aug 30 15:08:27 xserver influxd: runtime.sysMap(0xc460a70000, 0x100000, 0x0, 0x13568d8)
Aug 30 15:08:27 xserver influxd:        /usr/local/go/src/runtime/mem_bsd.go:85 +0x1ef
Aug 30 15:08:27 xserver influxd: runtime.(*mheap).sysAlloc(0x133d640, 0x100000, 0x0)
Aug 30 15:08:27 xserver influxd:        /usr/local/go/src/runtime/malloc.go:470 +0xd4
Aug 30 15:08:27 xserver influxd: runtime.(*mheap).grow(0x133d640, 0x8, 0x0)
Aug 30 15:08:27 xserver influxd:        /usr/local/go/src/runtime/mheap.go:907 +0x60
Aug 30 15:08:27 xserver influxd: runtime.(*mheap).allocSpanLocked(0x133d640, 0x8, 0x13568e8, 0x800)
Aug 30 15:08:27 xserver influxd:        /usr/local/go/src/runtime/mheap.go:820 +0x2d1
Aug 30 15:08:27 xserver influxd: runtime.(*mheap).alloc_m(0x133d640, 0x8, 0x7fffffff0101, 0x42460a)
Aug 30 15:08:27 xserver influxd:        /usr/local/go/src/runtime/mheap.go:686 +0x118
Aug 30 15:08:27 xserver influxd: runtime.(*mheap).alloc.func1()
Aug 30 15:08:27 xserver influxd:        /usr/local/go/src/runtime/mheap.go:753 +0x4d
Aug 30 15:08:27 xserver influxd: runtime.(*mheap).alloc(0x133d640, 0x8, 0x1131010101, 0x7fffffffec08)
Aug 30 15:08:27 xserver influxd:        /usr/local/go/src/runtime/mheap.go:752 +0x8a
Aug 30 15:08:27 xserver influxd: runtime.largeAlloc(0x10000, 0x7fffffff0101, 0x454b7e)
Aug 30 15:08:27 xserver influxd:        /usr/local/go/src/runtime/malloc.go:826 +0x94
Aug 30 15:08:27 xserver influxd: runtime.mallocgc.func1()
Aug 30 15:08:27 xserver influxd:        /usr/local/go/src/runtime/malloc.go:721 +0x46
Aug 30 15:08:27 xserver influxd: runtime.systemstack(0x0)
Aug 30 15:08:27 xserver influxd:        /usr/local/go/src/runtime/asm_amd64.s:409 +0x79
Aug 30 15:08:27 xserver influxd: runtime.mstart()
Aug 30 15:08:27 xserver influxd:        /usr/local/go/src/runtime/proc.go:1175

Hi.
We seem to be having the same issue

ts=2018-09-18T17:53:06.107192Z lvl=info msg="Failed to open shard" log_id=0AcSr6F0000 service=store trace_id=0AcSr6iW000 op_name=tsdb_open db_shard_id=13344 error="[shard 13344] cannot allocate memory"
ts=2018-09-18T17:53:06.107613Z lvl=info msg="Failed to open shard" log_id=0AcSr6F0000 service=store trace_id=0AcSr6iW000 op_name=tsdb_open db_shard_id=14733 error="[shard 14733] cannot allocate memory"
ts=2018-09-18T17:53:06.107834Z lvl=info msg="Failed to open shard" log_id=0AcSr6F0000 service=store trace_id=0AcSr6iW000 op_name=tsdb_open db_shard_id=16525 error="[shard 16525] cannot allocate memory"
ts=2018-09-18T17:53:06.108082Z lvl=info msg="Opened file" log_id=0AcSr6F0000 engine=tsm1 service=filestore path=/var/lib/influxdb/data/telegraf/medalliaRP/15543/000000011-000000001.tsm id=3 duration=108.968ms
ts=2018-09-18T17:53:06.108219Z lvl=error msg="Cannot read corrupt tsm file, renaming" log_id=0AcSr6F0000 engine=tsm1 service=filestore path=/var/lib/influxdb/data/telegraf/medalliaRP/15543/000000011-000000001.tsm id=3 error="cannot allocate memory"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x58 pc=0xaf6214]

goroutine 77103 [running]:
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*TSMReader).WithObserver(...)
    /go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/reader.go:273
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*FileStore).Open.func1(0xc684a91110, 0xc554f72420, 0x3, 0xc61d4720b8)
    /go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/file_store.go:535 +0xe74
created by github.com/influxdata/influxdb/tsdb/engine/tsm1.(*FileStore).Open
    /go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/file_store.go:510 +0x4a9
W0918 17:52:40.413782 193758 logging.cpp:91] RAW: Received signal SIGTERM from process 45216 of user 0; exiting

--- a/tsdb/engine/tsm1/file_store.go
+++ b/tsdb/engine/tsm1/file_store.go
@@ -547,6 +547,8 @@ func (f *FileStore) Open() error {
 					readerC <- &res{r: df, err: fmt.Errorf("cannot rename corrupt file %s: %v", file.Name(), e)}
 					return
 				}
+				readerC <- &res{r: df, err: fmt.Errorf("cannot read corrupt file %s: %v", file.Name(), err)}
+				return
 			}
 
 			df.WithObserver(f.obs)

